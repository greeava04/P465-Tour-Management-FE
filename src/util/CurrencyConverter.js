import React, { useEffect, useState, image } from 'react'
import Avatar from '@mui/material/Avatar';
import Button from '@mui/material/Button';
import CssBaseline from '@mui/material/CssBaseline';
import TextField from '@mui/material/TextField';
import FormControlLabel from '@mui/material/FormControlLabel';
import Checkbox from '@mui/material/Checkbox';
import Link from '@mui/material/Link';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import GoogleIcon from '@mui/icons-material/Google';
import Chip from '@mui/material/Chip';
import Divider from '@mui/material/Divider';
import LoginNavComponent from '../util/LoginNavComponent';

import InputLabel from '@mui/material/InputLabel';
import MenuItem from '@mui/material/MenuItem';
import FormControl from '@mui/material/FormControl';
import Select from '@mui/material/Select';


export default function CurrencyConverter() {
    const [input, setInput] = useState('');
    const [output, setOutput] = useState('')
    let [rates, setRates] = useState({});
    

    const handleChange = (event) => {
        setInput(event.target.value);
        console.log(input);
    };
    const handleChangeOut = (event) => {
        setOutput(event.target.value);
    };

    const handleSubmit = (event) => {
        event.preventDefault();
        const formData = new FormData(event.currentTarget);
        const formJson = Object.fromEntries(formData.entries());
        formJson.input = input;
        formJson.output = output;
        console.log(formJson);

        let outputValue = document.getElementById('outputCurrency');

        if (rates[input] && rates[input][output]) {
            console.log("Exists using prepopulated exchange")
            outputValue.innerText = Math.round(rates[input][output] * Number(formJson.inputCurrency) * 100) / 100

        } else {
            fetch(`http://owenhar1.asuscomm.com:3000/api/convertCurrency?base_currency=${input}&currencies=${output}`, {
                method: "GET",
                headers: {
                    "Content-Type": "application/json"
                }
            }).then((res) => res.json())
            .then((data) => {
                console.log(data)
                let tempRate = rates;
                tempRate[input] = {};
                tempRate[input][output] = data.data[output].value;
                setRates(tempRate);
                outputValue.innerText = Math.round(data.data[output].value * Number(formJson.inputCurrency) * 100) / 100;
            })
        }


    }


    return (
        <>
            <Box component="form" onSubmit={handleSubmit} noValidate sx={{ mt: 1 }}>
                <div class="row">
                    <InputLabel id="input-label">Input</InputLabel>
                    <Select
                        labelId="input"
                        id="input"
                        value={input}
                        label="Age"
                        onChange={handleChange}
                    >
                        <MenuItem value={"USD"}>USD</MenuItem>
                        <MenuItem value={"EUR"}>EUR</MenuItem>
                        <MenuItem value={"GBP"}>GBP</MenuItem>
                    </Select>
                    <TextField
                        margin="normal"
                        required
                        name="inputCurrency"
                        label="Input Currency"
                        type="number"
                        id="inputCurrency"
                    />
                </div>
                <div class="row">
                    <InputLabel id="output-label">Output</InputLabel>
                    <Select
                        labelId="ouput-label"
                        id="output"
                        value={output}
                        label="Age"
                        onChange={handleChangeOut}
                    >
                        <MenuItem value={"USD"}>USD</MenuItem>
                        <MenuItem value={"EUR"}>EUR</MenuItem>
                        <MenuItem value={"GBP"}>GBP</MenuItem>
                    </Select>
                    <div id="outputCurrency">
                        Output Currency
                    </div>
                </div>
                <Button
                    type="submit"
                    fullWidth
                    variant="contained"
                    sx={{ mt: 3, mb: 2, backgroundColor: '#2484BF' }}
                    style={{ backgroundColor: '#2484BF' }}
                >
                    Convert
                </Button>
            </Box>
        </>
    );

}