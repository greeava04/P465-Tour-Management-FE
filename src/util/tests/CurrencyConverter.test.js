import renderer from 'react-test-renderer';
import CurrencyConverter from '../CurrencyConverter';

it('renders page', () => {
    const component = renderer.create(
        <CurrencyConverter></CurrencyConverter>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})