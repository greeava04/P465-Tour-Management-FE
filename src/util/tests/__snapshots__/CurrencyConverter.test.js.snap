// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders page 1`] = `
<form
  className="MuiBox-root css-164r41r"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    class="row"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
      id="input-label"
    >
      Input
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      onClick={[Function]}
    >
      <div
        aria-controls=":r0:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="input input"
        className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="input"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </div>
      <input
        aria-hidden={true}
        aria-invalid={false}
        className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        tabIndex={-1}
        value=""
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            Age
          </span>
        </legend>
      </fieldset>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="inputCurrency"
        id="inputCurrency-label"
      >
        Input Currency
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="inputCurrency"
          name="inputCurrency"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="number"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Input Currency
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
      id="output-label"
    >
      Output
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      onClick={[Function]}
    >
      <div
        aria-controls=":r2:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="ouput-label output"
        className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="output"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </div>
      <input
        aria-hidden={true}
        aria-invalid={false}
        className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        tabIndex={-1}
        value=""
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            Age
          </span>
        </legend>
      </fieldset>
    </div>
    <div
      id="outputCurrency"
    >
      Output Currency
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1sbvkol-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#2484BF",
      }
    }
    tabIndex={0}
    type="submit"
  >
    Convert
  </button>
</form>
`;
